spec/rubocop/cop/rails/dot_separated_keys_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::DotSeparatedKeys do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense and corrects when translating keys with convertible scopes' do
+    expect_offense(<<~RUBY)
+      I18n.t :key, scope: [:one, :two]
+                   ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      I18n.translate :key, scope: [:one, :two]
+                           ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      t :key, scope: [:one, :two]
+              ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      translate :key, scope: [:one, :two]
+                      ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      t :key, scope: [:one, :two], default: 'Not here'
+              ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      I18n.t :key, scope: ['one', :two]
+                   ^^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      I18n.t 'key', scope: [:one, :two]
+                    ^^^^^^^^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      I18n.t :key, scope: 'one'
+                   ^^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+      I18n.t :key, scope: :one
+                   ^^^^^^^^^^^ Use the dot-separated keys instead of specifying the `:scope` option.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      I18n.t 'one.two.key'
+      I18n.translate 'one.two.key'
+      t 'one.two.key'
+      translate 'one.two.key'
+      t 'one.two.key', default: 'Not here'
+      I18n.t 'one.two.key'
+      I18n.t 'one.two.key'
+      I18n.t 'one.key'
+      I18n.t 'one.key'
+    RUBY
+  end
+
+  it 'does not register an offense when key is an array' do
+    expect_no_offenses(<<~RUBY)
+      t [:key1, :key2], scope: :one
+    RUBY
+  end
+
+  it 'does not register an offense when key is not a basic literal' do
+    expect_no_offenses(<<~RUBY)
+      t key1, scope: :one
+    RUBY
+  end
+
+  it 'does not register an offense when `scope` is an array containing non literals' do
+    expect_no_offenses(<<~RUBY)
+      t :key, scope: [:one, two]
+    RUBY
+  end
+
+  it 'does not register an offense when `scope` is not a string nor a symbol' do
```

Comment:
You mention that "`scope` is not a string". But `scope is an Array or a Sybmol in the examples.
Can you add an example for a string scope?

To my personal taste,
```ruby
t :record_not_found, scope: 'activerecord.errors.messages'
```
reads better than
```ruby
t scope: 'activerecord.errors.messages.record_not_found'
```
I'll gladly suggest an amendment to the Rails Style guide.

Reply:
```ruby
t :record_not_found, scope: 'activerecord.errors.messages'
```

will already be corrected to
```ruby
t 'activerecord.errors.messages.record_not_found'
```

Or am I missing something?

Reply:
https://github.com/rubocop/rails-style-guide/pull/299

Reply:
> am I missing something?

Yes, an example for this case 😄

Reply:
Updated with that change and a new test case.